migrations/entitlements/migration_test.go

```diff
@@ -2108,3 +2117,85 @@ func TestMigrateDictOfWithTypeValueKey(t *testing.T) {
 		ref.Authorization,
 	)
 }
+
+func TestConvertDeprecatedStaticTypes(t *testing.T) {
+
+	t.Parallel()
+
+	test := func(ty interpreter.PrimitiveStaticType) {
+
+		t.Run(ty.String(), func(t *testing.T) {
+			t.Parallel()
+
+			inter := NewTestInterpreter(t)
+			value := interpreter.NewUnmeteredCapabilityValue(
+				1,
+				interpreter.AddressValue(common.ZeroAddress),
+				interpreter.NewReferenceStaticType(
+					nil,
+					interpreter.UnauthorizedAccess,
+					ty,
+				),
+			)
+
+			result, err := ConvertValueToEntitlements(inter, value)
+			require.Error(t, err)
+			assert.ErrorContains(t, err, "cannot migrate deprecated type")
+			require.Nil(t, result)
+		})
+	}
+
+	for ty := interpreter.PrimitiveStaticType(1); ty < interpreter.PrimitiveStaticType_Count; ty++ {
+		if !ty.IsDefined() || !ty.IsDeprecated() { //nolint:staticcheck
+			continue
+		}
+
+		test(ty)
+	}
+}
+
+func TestConvertMigratedAccountTypes(t *testing.T) {
+
+	t.Parallel()
+
+	test := func(ty interpreter.PrimitiveStaticType) {
+
+		t.Run(ty.String(), func(t *testing.T) {
+			t.Parallel()
+
+			inter := NewTestInterpreter(t)
+			value := interpreter.NewUnmeteredCapabilityValue(
+				1,
+				interpreter.AddressValue(common.ZeroAddress),
+				interpreter.NewReferenceStaticType(
+					nil,
+					interpreter.UnauthorizedAccess,
+					ty,
+				),
+			)
+
+			newValue, err := account_type.NewAccountTypeMigration().Migrate(
+				interpreter.AddressPath{
+					Address: common.ZeroAddress,
+					Path:    interpreter.EmptyPathValue,
+				},
+				value,
+				inter,
+			)
+			require.NoError(t, err)
+			require.NotNil(t, newValue)
+
+			result, err := ConvertValueToEntitlements(inter, newValue)
+			require.NoError(t, err)
+			require.Nilf(t, result, "expected no migration, but got %s", result)
```

Comment:
@dsainati1 Where in the entitlements migration do we need to add the check if the given `sema.Type` to be migrated should be ignored because it got migrated from a deprecated primitive static type (e.g. `&Account` should not be migrated)?

Reply:
I think in `ConvertToEntitledType` you should be able to just return the input unchanged if it is a deprecated type that we should ignore.

Reply:
I had initially tried that at the top and it didn't work, but realized that it must be specifically added in the reference type case which adds the authorization